### PR TITLE
fix: audit log filtering & sorting

### LINKED ISSUE #2304: oldest_first=True in the audit logs returns newest_first
### Summary

When trying to get oldest_first, it returns newest firsts

### Reproduction Steps

I am trying to get the oldest_first in the audit logs:
```python
                async for entry in guild.audit_logs(limit=None, after=object_id,
                                                    oldest_first=True, action=discord.AuditLogAction.member_role_update):
                    print(f'{entry.created_at}')
```
But it is returning the most recent first ...
```
2023-12-28 19:49:21.731000+00:00
2023-12-28 19:49:18.526000+00:00
2023-12-28 19:49:14.620000+00:00
2023-12-28 19:49:11.040000+00:00
```

### Minimal Reproducible Code

```python
async for entry in guild.audit_logs(limit=None, after=object_id,
                                                    oldest_first=True, action=discord.AuditLogAction.member_role_update):
                    print(f'{entry.created_at}')
```

### Expected Results

```
2023-12-28 19:49:11.040000+00:00
2023-12-28 19:49:14.620000+00:00
2023-12-28 19:49:18.526000+00:00
2023-12-28 19:49:21.731000+00:00
```

### Actual Results

```
2023-12-28 19:49:21.731000+00:00
2023-12-28 19:49:18.526000+00:00
2023-12-28 19:49:14.620000+00:00
2023-12-28 19:49:11.040000+00:00
```

### Intents

moderation, members, guilds

### System Information

- Python v3.11.1-final
- py-cord-dev v2.5.None-candidate
    - py-cord-dev pkg_resources: v2.5.0rc5
- aiohttp v3.8.5
- system info: Windows 10 10.0.22631

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

### LINKED ISSUE #2445: audit_logs not fetching more than 100 logs
### Summary

audit_logs not fetching more than 100 logs

### Reproduction Steps

async for entry in guild.audit_logs(limit=None, after=discord.Object(id=0),                                action=discord.AuditLogAction.member_role_update):
  ...

### Minimal Reproducible Code

```python
async for entry in guild.audit_logs(limit=None, after=discord.Object(id=0),                                action=discord.AuditLogAction.member_role_update):
  ...
```

### Expected Results

Fetch all the audit logs in batches:
fetch 1-100, 101-202, ... etc
but in the iterator, we only see them coming one at a time
without to have to do multiple calls

### Actual Results

Only fetch the 100 lastest audit logs

### Intents

    intents.members = True     intents.guilds = True     intents.moderation = True     intents.voice_states = True

### System Information

- Python v3.11.1-final
- py-cord v2.5.1-final
- aiohttp v3.8.6
- system info: Windows 10 10.0.22631

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

The problem migh have been introduced by the after param being fixed: 
Fixed AuditLogIterator not respecting the after parameter. ([#2295](https://github.com/Pycord-Development/pycord/issues/2295))

### PR DESCRIPTION
**Pull request by Lulalaby**
## Summary

This pull requests aims to remove the braindead logic for "sorting" audit logs.
We get the data already based on param combo.

Closes #2304 
Fixes #2445

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Blue-Robin-Taken — APPROVED** [review]

**Comment by Lulalaby** [comment]
@Blue-Robin-Taken im not sure if that even works lmfao, did u tested?

File: discord/iterators.py [hunk 1]
```diff
@@ -444,46 +442,28 @@ def __init__(
...
 
-        if self.reverse:
-            self._strategy = self._after_strategy
-            if self.before:
-                self._filter = lambda m: int(m["id"]) < self.before.id
-        else:
-            self._strategy = self._before_strategy
-            if self.after and self.after != OLDEST_OBJECT:
-                self._filter = lambda m: int(m["id"]) > self.after.id
+        self._strategy = self._strategy_exec
```
Comment (Dorukyum) [inline discord/iterators.py:445]: If there's a single strategy, there's no need to store it as a "strategy", it can be executed directly.

**Review by yoggys — APPROVED** [review]
I just tested it and it works fine

**Comment by Dorukyum** [comment]
This should fix #2445.

**Comment by Lulalaby** [comment]
can someone test this again please

File: discord/iterators.py [hunk 2]
```diff
@@ -501,46 +499,28 @@ def __init__(
...
 
-    async def _before_strategy(self, retrieve):
+    async def _strategy_exec(self, retrieve):
```
Comment (plun1331) [inline discord/iterators.py]: I believe this is what doruk meant:
```suggestion

    async def _strategy(self, retrieve):
```

Comment (Lulalaby) [reply discord/iterators.py]: That doesn't exist anymore

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**